File: ControlLab/src/main/java/org/marsroboticsassociation/controllab/trajectory/TrajectorySvgSeries.java
package org.marsroboticsassociation.controllab.trajectory;

import org.marsroboticsassociation.controllib.motion.PolynomialCurveSegment;

import java.awt.BasicStroke;
import java.awt.Color;
import java.util.List;

record TrajectorySvgSeries(
        String label,
        List<PolynomialCurveSegment> segments,
        Color color,
        float strokeWidth,
        BasicStroke stroke) {}

File: ControlLab/src/main/kotlin/org/marsroboticsassociation/controllab/trajectory/TrajectorySvgSeries.kt
package org.marsroboticsassociation.controllab.trajectory

import java.awt.BasicStroke
import java.awt.Color
import org.marsroboticsassociation.controllib.motion.TrajectoryCurveSegment

@JvmRecord
data class TrajectorySvgSeries(
    val label: String,
    val segments: List<TrajectoryCurveSegment>,
    val color: Color,
    val strokeWidth: Float,
    val stroke: BasicStroke?,
)
